Examples/Algorithms/Geant4/include/ActsExamples/Geant4/EventStore.hpp

```diff
@@ -0,0 +1,91 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021-2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Material/MaterialInteraction.hpp"
+#include "ActsExamples/EventData/SimHit.hpp"
+#include "ActsExamples/EventData/SimParticle.hpp"
+#include "ActsExamples/Framework/DataHandle.hpp"
+#include "ActsFatras/EventData/Barcode.hpp"
+
+#include <memory>
+#include <mutex>
+#include <set>
+#include <unordered_map>
+#include <vector>
+
+#include "G4Types.hh"
+
+namespace ActsExamples {
+
+class WhiteBoard;
+
+/// Common event store for all Geant4 related sub algorithms
```

Comment:
I would maybe consider only store a `shared_ptr` to this obj in the algorithm, and don't add an additional layer until it is needed.